src/celeritas/io/NeutronXsReader.cc

```diff
@@ -0,0 +1,99 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/io/NeutronXsReader.cc
+//---------------------------------------------------------------------------//
+#include "NeutronXsReader.hh"
+
+#include <fstream>
+#include <vector>
+
+#include "corecel/Assert.hh"
+#include "corecel/Types.hh"
+#include "corecel/io/Logger.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/sys/Environment.hh"
+#include "celeritas/Units.hh"
+#include "celeritas/io/ImportLivermorePE.hh"
+#include "celeritas/io/ImportPhysicsVector.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the reader using the G4PARTICLEXSDATA environment variable to get
+ * the path to the data.
+ */
+NeutronXsReader::NeutronXsReader()
+{
+    std::string const& dir = celeritas::getenv("G4PARTICLEXSDATA");
+    CELER_VALIDATE(!dir.empty(),
+                   << "environment variable G4PARTICLEXSDATA is not defined"
+                      " (needed to locate Livermore data)");
+    path_ = dir + "/neutron";
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the reader with the path to the directory containing the data.
+ */
+NeutronXsReader::NeutronXsReader(char const* path) : path_(path)
+{
+    CELER_EXPECT(!path_.empty());
+    if (path_.back() == '/')
+    {
+        path_.pop_back();
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Read the data for the given element.
+ */
+NeutronXsReader::result_type
+NeutronXsReader::operator()(AtomicNumber atomic_number) const
+{
+    CELER_EXPECT(atomic_number && atomic_number < AtomicNumber{92});
```

Comment:
There's no scattering data for uranium and transuranics? Or it just requires another model? Maybe it would be better to remove this hard-coded value and let it fail if there's no file at the requested element?

Reply:
Actually I mistakenly dropped the uranium when excluding transuranium elements as their neutron elastic cross sections do not exist (in G4PARTICLEXS data set) even though the CHIPS model itself uses a set of default values of parameters for them with a warning message (so, do not know whether it is meaningful or not).  Anyway, I remove the condition and make it fail when reading unavailable data file.